src/runtime/converter.cs

```diff
@@ -133,6 +134,22 @@ internal static IntPtr ToPython(object value, Type type)
                 return result;
             }
 
+            if (value is IList && value.GetType().IsGenericType)
+            {
+                using (var resultlist = new PyList())
+                {
+                    foreach (object o in (IEnumerable)value)
```

Comment:
I left that alone on purpose as it wasn't necessary to modify for what this `pr` is trying to do. 

The correct thing there is probably to delete that section of code altogether though (or delete `typecode.Object`). The remaining typecodes are `Empty`, `DBNull`, `Decimal` and `DateTime` and i don't think any of those should be converted to `PyList`. 
Also, I can't think of a situation in which the `default` branch of the switch would be hit because we have `Typecode.Object` defined; It would always be called instead of the `default`. This [example](http://stackoverflow.com/a/7542810/5208670) is probably closer to how it should be written. 
Either way though, all those changes aren't needed for the purposes of fixing the `numpy arrays` which is why I didn't include them.